### DISCUSSION TITLE
router modified through the blocker is not being parsed by searchValidation.

### DISCUSSION THREAD

**Discussion by dm0114** [TanStack/router#1913] [Q&A]
Why is the router modified through the blocker not being parsed by searchValidation? It doesn't seem to be thrown by beforeLoad either. I want to know why there are no searchParams in the URL and why it just shows 'posts'.

```typescript
//example

/** ComponentA */
useBlocker({
  blockerFn: VoidFunction,
  condition: isDirty
})
<Link href="/posts" >posts</>


/** PostsRoute */
export cont Route = ~
component: Posts,
validateSearch: (search) => zodSchema.parse(search)

/** Posts */
const searchParams = useSearch() // undefined
```

**Comment by schiller-manuel** [comment]
can you please provide a minimal complete example by forking one of the existing examples on stackblitz?
e.g. https://tanstack.com/router/latest/docs/framework/react/examples/kitchen-sink-react-query-file-based
